components/dashboard/src/components/UsageBasedBillingConfig.tsx

```diff
@@ -92,7 +92,17 @@ export default function UsageBasedBillingConfig({ attributionId }: Props) {
             setPendingStripeSubscription(pendingSubscription);
             window.localStorage.setItem(localStorageKey, JSON.stringify(pendingSubscription));
             try {
-                await getGitpodService().server.subscribeToStripe(attributionId, setupIntentId);
+                // Pick a good initial value for the Stripe usage limit (1000 * team_size)
+                // FIXME: Should we ask the customer to confirm or edit this default limit?
+                let limit = 1000;
+                const attrId = AttributionId.parse(attributionId);
+                if (attrId?.kind === "team") {
+                    const members = await getGitpodService().server.getTeamMembers(attrId.teamId);
+                    limit = 1000 * members.length;
+                }
+                await getGitpodService().server.subscribeToStripe(attributionId, setupIntentId, limit);
+                const newLimit = await getGitpodService().server.getUsageLimit(attributionId);
```

Comment:
Why would we get the usage limit now that we've set it to a given value (which we know) and the call hasn't failed because it proceeded to the next expression?

Reply:
Instead of assuming we know the current limit (based on our current understanding of the backend), I thought it's better to leave the backend (i.e. Cost Center logic) be the single authority on what limits are legal / acceptable / and when they should be reset / to what.

I feel like not asking, and proceeding on assumptions would make this code more fragile.

Reply:
Shall we return the new cost center (and the assigned spending limit) from the `subscribeToStripe` call instead? That way, we push this logic down into the backend rather than having the client deal with this logic.

Reply:
@easyCZ I like this idea! (Returning a CostCenter object to the front-end, or maybe just the usage limit?)

Reply:
For this, I'll extend the `SetCostCenter` API to actually return the updated CostCenter such that you don't have to do Update followed be a Get

Reply:
Refactored to not do a Get after Set ✅ thanks!